pm/recipient.go

```diff
@@ -363,16 +361,21 @@ func (r *recipient) updateSenderNonce(rand *big.Int, ticket *Ticket) error {
 	defer r.senderNoncesLock.Unlock()
 
 	randStr := rand.String()
-	sn, ok := r.senderNonces[randStr]
-	if ok && ticket.SenderNonce <= sn.nonce {
-		return errors.Errorf("invalid ticket senderNonce sender=%v nonce=%v highest=%v", ticket.Sender.Hex(), ticket.SenderNonce, sn.nonce)
+	nonceToExpBlock, randKeySeen := r.senderNonces[randStr]
+	if randKeySeen {
+		_, nonceSeen := nonceToExpBlock.Get(ticket.SenderNonce)
+		if nonceSeen {
+			return errors.Errorf("invalid ticket senderNonce: already seen sender=%v nonce=%v", ticket.Sender.Hex(), ticket.SenderNonce)
+		}
+	} else {
+		r.senderNonces[randStr] = orderedmap.NewOrderedMap()
+	}
+	// add new nonce
+	r.senderNonces[randStr].Set(ticket.SenderNonce, ticket.ParamsExpirationBlock)
+	// delete first seen nonce, if ordered map exceeds max size
+	if r.senderNonces[randStr].Len() > maxSenderNonces {
+		r.senderNonces[randStr].Delete(r.senderNonces[randStr].Front().Key)
```

Comment:
Seems simpler to just return an error if the max size is hit. If we delete the first seen nonce, then that means that in theory B could re-use that deleted nonce after the max size is hit and if we return an error if the max size is hit we at least wouldn't need to reason about this scenario. This would mean that once the max size is hit, B would *need* a new set of ticket params with a fresh `recipientRandHash`, but as mentioned in the original issue at the moment this should happen after B receives a response for each transcoded segment.

If we don't need to delete the first seen senderNonce value I think that would also simplify the data struct impl s.t. we might not need to depend on the custom ordered map package.

WDYT?